WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -90,24 +90,31 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 					trackedWallets.Remove(finishedCoinJoin.Wallet.WalletName);
 					finishedCoinJoin.CancellationTokenSource.Dispose();
 
-					var finishedCoinJoinTask = finishedCoinJoin.CoinJoinTask;
-					if (finishedCoinJoinTask.IsCompletedSuccessfully)
+					try
 					{
-						Logger.LogInfo("Coinjoin client finished successfully!");
-					}
-					else if (finishedCoinJoinTask.IsFaulted)
-					{
-						if (finishedCoinJoinTask.Exception?.InnerException is InvalidOperationException ioe)
+						var success = await finishedCoinJoin.CoinJoinTask.ConfigureAwait(false);
+						if (success)
+						{
+							Logger.LogInfo("Coinjoin client finished successfully!");
+						}
+						else
 						{
-							await Task.Delay(TimeSpan.FromSeconds(5), stoppingToken).ConfigureAwait(false);
-							continue;
+							Logger.LogInfo("Coinjoin client finished with error. Transaction not broadcasted.");
 						}
-						Logger.LogError(finishedCoinJoinTask.Exception!);
 					}
-					else if (finishedCoinJoinTask.IsCanceled)
+					catch (InvalidOperationException ioe)
+					{
+						Logger.LogError(ioe);
+						await Task.Delay(TimeSpan.FromSeconds(5), stoppingToken).ConfigureAwait(false);
```

Comment:
Could you explain why it is useful to sleep 5 seconds?

More importantly, when stoppingToken is actually canceled, the rest of `finishedCoinJoins` is not disposed. AFAIK we try to dispose every time.

Reply:
> Could you explain why it is useful to sleep 5 seconds?

No, IDK the reason behind that delay. @molnard ?

> More importantly, when stoppingToken is actually canceled, the rest of finishedCoinJoins is not disposed.

Good catch!

Reply:
(It seems this is not resolved but is marked so)

Reply:
Yes, it was marked by mistake.